From 328aa25761852dda49afe9e6dc5bf9ed3784046e Mon Sep 17 00:00:00 2001
From: Marek Maslanka <mm@semihalf.com>
Date: Wed, 30 Nov 2022 13:07:22 +0000
Subject: [PATCH] Revert "UPSTREAM: include/linux/mm_inline.h: fold __update_lru_size() into its sole caller"

This reverts commit 992fba2af9119f69b306d149eea55cd1996876ae.

BUG=b:213475212
TEST=

Change-Id: I62f9bd6527eaf1feb7e3fce11f272674ad17b8e5
Signed-off-by: Marek Maslanka <mm@semihalf.com>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/4068019
Reviewed-by: Yu Zhao <yuzhao@chromium.org>
---

diff --git a/include/linux/mm_inline.h b/include/linux/mm_inline.h
index 355ea1ee..7183c7a 100644
--- a/include/linux/mm_inline.h
+++ b/include/linux/mm_inline.h
@@ -24,7 +24,7 @@
 	return !PageSwapBacked(page);
 }
 
-static __always_inline void update_lru_size(struct lruvec *lruvec,
+static __always_inline void __update_lru_size(struct lruvec *lruvec,
 				enum lru_list lru, enum zone_type zid,
 				int nr_pages)
 {
@@ -33,6 +33,13 @@
 	__mod_lruvec_state(lruvec, NR_LRU_BASE + lru, nr_pages);
 	__mod_zone_page_state(&pgdat->node_zones[zid],
 				NR_ZONE_LRU_BASE + lru, nr_pages);
+}
+
+static __always_inline void update_lru_size(struct lruvec *lruvec,
+				enum lru_list lru, enum zone_type zid,
+				int nr_pages)
+{
+	__update_lru_size(lruvec, lru, zid, nr_pages);
 #ifdef CONFIG_MEMCG
 	mem_cgroup_update_lru_size(lruvec, lru, zid, nr_pages);
 #endif
